@xen-orchestra/lite/src/components/Modal.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react'
+import { FormattedMessage } from 'react-intl'
+import { withState } from 'reaclette'
+
+import Button from './Button'
+
+interface GeneralPropsModal {
+  message: JSX.Element
+  title: JSX.Element
+}
+
+interface PropsModal extends GeneralPropsModal {
+  onReject: (reason: unknown) => void
+  onSuccess: (value: unknown) => void
+}
+
+export const confirm = ({ message, title }: GeneralPropsModal) =>
+  new Promise((resolve, reject) => modal({ message, onReject: reject, onSuccess: resolve, title }))
+
+const modal = ({ message, onReject, onSuccess, title }: PropsModal) => {
+  if (instance === undefined) {
+    throw new Error('No modal instance')
+  }
+
+  instance.state.message = message
+  instance.state.onReject = onReject
+  instance.state.onSuccess = onSuccess
+  instance.state.showModal = true
+  instance.state.title = title
+}
+
+const STYLES_MODAL: React.CSSProperties = {
+  backgroundColor: 'lightgrey',
+  border: '1px solid black',
+  left: '50%',
+  position: 'absolute',
+  top: '20%',
+  transform: 'translate(-50%,-50%)',
+}
+
+interface ParentState {}
+
+interface State {
+  message?: JSX.Element
+  onReject?: (reason: unknown) => void
+  onSuccess?: (value: unknown) => void
+  showModal: boolean
+  title?: JSX.Element
+}
+
+interface Props {}
+
+interface ParentEffects {}
+
+interface Effects {
+  _closeModal: () => void
+  _reject: () => void
+  _success: () => void
+}
+
+interface Computed {}
+
+let instance: EffectContext<State, Props, Effects, Computed, ParentState, ParentEffects> | undefined
+const Modal = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      message: undefined,
+      onReject: undefined,
+      onSuccess: undefined,
+      showModal: false,
+      title: undefined,
+    }),
+    effects: {
+      initialize: function () {
+        if (instance !== undefined) {
+          throw new Error('Modal is a singelton')
+        }
+        instance = this
+      },
+      finalize: function () {
+        instance = undefined
+      },
+      _closeModal: function () {
+        this.state.showModal = false
+      },
+      _reject: function () {
+        this.state.onReject!('Normal reject')
```

Comment:
What's the purpose of the argument?

Reply:
`reject` and `resolve` are two function that getting one parameter. This one are forwarded to `.then(response)/.catch(error)`.

When we close the modal with the cancel button, it will trigger an error, because the promise is rejected.
So i think that is a good idea to say that an error is thrown, but  that is the normal behaviour

Reply:
Then you could reject with `undefined` and it would mean that the modal was closed. That's the behavior we currently have in XO.